src/antidote.erl

```diff
@@ -126,33 +126,59 @@ read_objects(Objects, TxId) ->
 -spec update_objects([{bound_object(), op_name(), op_param()}], txid())
                     -> ok | {error, reason()}.
 update_objects(Updates, TxId) ->
-    %% Execute each update as in an interactive transaction
-    Results = lists:map(
-                fun({{Key, Type, Bucket}, Op, OpParam}) ->
-                        case clocksi_iupdate(TxId, {Key, Bucket}, Type,
-                                             {Op, OpParam}) of
-                            ok -> ok;
-                            {error, Reason} ->
-                                lager:debug("Update failed. Reason : ~p",[Reason]),
-                                error
-                        end
-                end, Updates),
-    case lists:member(error, Results) of
-       true -> {error, update_failed}; %% TODO: Capture the reason for error
-       false -> ok
+    {_, _, CoordFsmPid} = TxId,
+    Operations = lists:map(fun(Update) ->
+                                {Key, Type, Bucket, Op} = case Update of
+                                    %% the following checks the format of the
+                                    %% operation for compatibility with
+                                    %% some systests that call updates with
+                                    %% {Operation, Params} (as a single parameter),
+                                    %% and Operation, Params (two separate parameters).
+                                    {{K, T, B}, O} ->
```

Comment:
I think fixing the tests would be more sensible..., if not now then in a test refactoring step.
But why has this not been an issue before?